intellij/src/main/kotlin/motif/intellij/MotifProjectComponent.kt

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2018-2019 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package motif.intellij
+
+import com.intellij.codeInsight.daemon.LineMarkerProviders
+import com.intellij.lang.Language
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.components.ProjectComponent
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.progress.ProgressManager
+import com.intellij.openapi.progress.Task
+import com.intellij.openapi.project.DumbService
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.wm.ToolWindow
+import com.intellij.openapi.wm.ToolWindowAnchor
+import com.intellij.openapi.wm.ToolWindowManager
+import com.intellij.psi.PsiClass
+import com.intellij.psi.PsiElement
+import com.intellij.ui.content.Content
+import com.intellij.ui.content.ContentFactory
+import motif.core.ResolvedGraph
+import motif.intellij.ScopeHierarchyUtils.Companion.isMotifScopeClass
+import motif.intellij.provider.ScopeNavigationLineMarkerProvider
+import motif.intellij.ui.MotifErrorPanel
+import motif.intellij.ui.MotifPanel
+
+class MotifProjectComponent(val project: Project) : ProjectComponent {
+
+    companion object {
+        const val TOOL_WINDOW_ID: String = "Motif"
+        const val TOOL_WINDOW_TITLE: String = "Scopes"
+        const val TAB_NAME_ERRORS: String = "Errors"
+        const val TAB_NAME_SCOPES: String = "Scopes"
+        const val LABEL_GRAPH_REFRESH: String = "Refreshing Motif Graph"
+        const val LABEL_GRAPH_INIT: String = "Initializing Motif Graph"
+
+        fun getInstance(project: Project): MotifProjectComponent {
+            return project.getComponent(MotifProjectComponent::class.java)
+        }
+    }
+
+    private val graphFactory: GraphFactory by lazy { GraphFactory(project) }
+    private lateinit var toolWindow: ToolWindow
+    private lateinit var panel: MotifPanel
```

Comment:
Let's just make the panel nullable. Then in `onSelectedScope` use Kotlin's ["safe call"](https://kotlinlang.org/docs/reference/null-safety.html#safe-calls):

```kotlin
panel?.setSelectedScope(element)
```